ethereum/contracts/Splitter.sol

```diff
@@ -2,72 +2,27 @@ pragma solidity >= 0.5.7;
 pragma experimental ABIEncoderV2;
 
 contract Splitter {
-  struct User {
-    string name;
-    address account;
-  }
-  User[] users;
   mapping(address => bool) accounts;
   address owner;
 
   constructor() public {
     owner = msg.sender;
   }
 
-  function registerUser(string memory name) public {
-    require(!accounts[msg.sender], 'A given address is already registered');
-
-    User memory newUser = User({
-      name: name,
-      account: msg.sender
-    });
-    users.push(newUser);
-    accounts[msg.sender] = true;
-  }
-
-  event EtherSplit();
-
-  function splitEther() public payable {
-    require(accounts[msg.sender], 'A given address is not registered');
-    require(users.length > 2, 'Please wait until at least 3 users are registered');
-    require(msg.value > 0, 'A given value should be bigger than 0 (wei)');
-    require(msg.value / (users.length - 1) > 0, 'A given value is too small to be sent to others');
-
-    uint value = msg.value / (users.length - 1);
-
-    for (uint i = 0; i < users.length; i++) {
-      User storage user = users[i];
-      if (msg.sender != user.account) {
-        address payable recipient = address(uint160(user.account));
-        recipient.transfer(value);
-      }
-    }
+  event Transfer(address from, bool status, uint balance1, uint balance2);
 
-    emit EtherSplit();
-  }
-
-  function getAllUsers() public view returns (User[] memory) {
-    return users;
-  }
-
-  function deleteUser() public {
-    require(accounts[msg.sender], 'A given account does not exist');
-
-    accounts[msg.sender] = false;
-    for (uint i = 0; i < users.length; i++) {
-      if (msg.sender == users[i].account) {
-        delete users[i];
-      }
-    }
-  }
+  function sendEther(address[] memory receivers) public payable {
+    // One transfer call needs 2300 gas
+    require(msg.sender.balance + 4600 > msg.value, 'A sender should have enough balance');
+    require(receivers.length == 2, 'The number of receivers should be 2');
+    require(msg.value % 2 == 0, 'The value should be evenly divisible');
+    require(msg.value > 0, 'A given value should be bigger than 0');
+    require(msg.sender != receivers[0] && msg.sender != receivers[1], 'A sender should not be one of receivers');
 
-  // only for unit test
-  function deleteAllUsers() public {
-    require(msg.sender == owner, 'Delete all users can be run by owner only');
+    uint value = msg.value / 2;
+    address(uint160(receivers[0])).transfer(value);
```

Comment:
why do you convert address=>uint160=>address? Is it because the function arguments was not marked `payable`?